smartapps/ady624/webcore-piston.src/webcore-piston.groovy

```diff
@@ -3388,140 +3262,140 @@ private long vcmd_httpRequest(rtData, device, params) {
 					setSystemVariableValue(rtData, "\$httpContentType", response.contentType)
 					setSystemVariableValue(rtData, "\$httpStatusCode", response.status)
 					setSystemVariableValue(rtData, "\$httpStatusOk", (response.status >= 200) && (response.status <= 299))
-                    def binary = false
-                    def mediaType = response.contentType.toLowerCase()
-                    switch (mediaType) {
-                    	case 'image/jpeg':
-                    	case 'image/png':
-                    	case 'image/gif':
-                        	binary = true
-                    }
+					def binary = false
+					def mediaType = response.contentType.toLowerCase()
+					switch (mediaType) {
+						case 'image/jpeg':
+						case 'image/png':
+						case 'image/gif':
+							binary = true
+					}
 					if ((response.status == 200) && response.data && !binary) {
 						try {
 							rtData.response = response.data instanceof Map ? response.data : (LinkedHashMap) new groovy.json.JsonSlurper().parseText(response.data)
 						} catch (all) {
-                        	rtData.response = response.data
+							rtData.response = response.data
 						}
 					} else {
-                    	rtData.response = null
-                        if (response.data && (response.data instanceof java.io.ByteArrayInputStream)) {
-	                        rtData.mediaType = mediaType
-    	                    rtData.mediaData = response.data.getBytes()
-                        } else {
-	                        rtData.mediaType = null
-    	                    rtData.mediaData = null
-                        }
-                        rtData.mediaUrl = null;
-                    }
+						rtData.response = null
+						if (response.data && (response.data instanceof java.io.ByteArrayInputStream)) {
+							rtData.mediaType = mediaType
+							rtData.mediaData = response.data.getBytes()
+						} else {
+							rtData.mediaType = null
+							rtData.mediaData = null
+						}
+						rtData.mediaUrl = null;
+					}
 				}
+				pause(400)
```

Comment:
Why the thread lock here?

Reply:
I viewed pushing a lot of http requests as another loop lock up scenario.  This again was to let something else in (or the UI), if you had a problem of these happening quickly.   It is possible the other loop control would handle this.  Unfortunately there is not an easily accessible test suite use.